### PR TITLE
Clear temporary buffer when filtering

### PR DESCRIPTION
**Pull request by ethomson**
Fixes #3009

### CODE REVIEW

File: tests/checkout/crlf.c [hunk 1]
```diff
@@ -347,3 +347,24 @@ void test_checkout_crlf__autocrlf_input_text_auto_attr(void)
...
+	git_checkout_head(g_repo, &opts);
+
+	check_file_contents("./crlf/test1.txt", "");
+
+	if (GIT_EOL_NATIVE == GIT_EOL_LF)
+		check_file_contents("./crlf/test2.txt", "test2.txt's content\n");
+	else
+		check_file_contents("./crlf/test2.txt", "test2.txt's content\r\n");
+
+	/* this will fail with core.autocrlf true and running it on Windows */
```
Comment (jacquesg) [inline tests/checkout/crlf.c]: This comment is no longer valid after the fix, it will pass as expected.

File: tests/checkout/crlf.c [hunk 2]
```diff
@@ -347,3 +347,24 @@ void test_checkout_crlf__autocrlf_input_text_auto_attr(void)
...
 }
+
+void test_checkout_crlf__can_write_empty_file(void)
+{
+	git_checkout_options opts = GIT_CHECKOUT_OPTIONS_INIT;
+	opts.checkout_strategy = GIT_CHECKOUT_FORCE;
+
+	cl_repo_set_bool(g_repo, "core.autocrlf", true);
+
+	git_repository_set_head(g_repo, "refs/heads/bad-content");
```
Comment (jacquesg) [inline tests/checkout/crlf.c]: Should probably rename this to something more meaningful like 'empty-files'

**Comment by jacquesg** [comment]
This looks good barring the nonsensical names I chose when creating the test. Maybe want to squash my 2 commits into 1 as the first commit did not cater for CRLF?

**Comment by jacquesg** [comment]
@ethomson Can we get this merged? (Even if we submit another PR to fix the really minor nitpick comments I had about the test meta files) This is really a blocker on Windows at this point in time.

**Comment by ethomson** [comment]
Thanks @jacquesg !

**Comment by jacquesg** [comment]
:+1: thanks for the fix!

**Pull request merged**